### ISSUE TITLE
Parameters for smaller images

### ISSUE THREAD

**Issue by SHEKOLDA** [rbgirshick/py-faster-rcnn#371]
Hi, so I'm trying to train Faster R-CNN on a dataset of pretty small images (85x29). As far as I'm aware, when given such an image, Faster R-CNN expands it to be 1000x600, then this image goes through the neural network. But with default parameters for stride and base anchor size, which are 16 pixels (on the expanded image), isn't it like producing 9 identicall 1x1 ancors for each pixel of the original image (since it is more than 200 times smaller, 16 pixels on the large image equals less than a pixel on the small one)?

Do I get it right? And if so, should I greatly increase stride and anchor size for it to work with smaller images? Or is it better to change those 1000x600 sizes and not expand the image so much? Thank you.

**Comment by haihaoshen** [comment]
I am also interested in the proposal. From my viewpoint, you can resize the image to smaller size. RCNN actually accepts the image of 224 x 224.

**Comment by absorbguo** [comment]
Hi,@haihaoshen 
In faster-rcnn ,the network did not resize the input image to 224*224. Indeed,with the ROI pooling layer the network could process the raw input without resizing it.

**Comment by haihaoshen** [comment]
Yes, but for scoring, no matter the image size is resized (to 224 x 224) or
not, the result is same.

**Comment by acpn** [comment]
I was with the same problem, but when I change the parameter of images sizes in config.py script inside  lib/fastrcnn the detections were very high and solve my problem.

**Comment by haihaoshen** [comment]
Thanks for note. Which parameter in config.py did you change? I saw the
below two may relate to input image size.

__C.TRAIN.SCALES = (600,)

__C.TRAIN.MAX_SIZE = 1000

**Comment by acpn** [comment]
Exactly that, remember of change in test stage too in the same script.

**Comment by haihaoshen** [comment]
Thanks for your confirmation.

**Comment by harjatinsingh** [comment]
Did only changing the image input size work for detecting smaller objects ?

**Comment by absorbguo** [comment]
@harjatinsingh I don't think only resize the input image would work a lot for detetecting small objects because key parameters of detection are anchor size and featstride in the region proposal networks.

**Comment by harjatinsingh** [comment]
@absorbguo Thank you for the reply. I have understood the use of anchor size and fixed it, however i am not able to understand the use of feat_stride parameter. Can you point me in the right direction to learn about the meaning of feat_stride ?

**Comment by absorbguo** [comment]
@harjatinsingh In short, smaller featstride represents denser anchor boxes, larger featstride represents sparser anchor boxes. Denser anchor boxes means predicting more proposal in the same area compared with sparser anchor boxes.You can review the generator.py for more details.

**Comment by harjatinsingh** [comment]
@absorbguo Thank you.

**Comment by gentlebreeze1** [comment]
can not only change __C.TEST.SCALES = (1080,) __C.TEST.MAX_SIZE = 1920   what else should i change? @absorbguo
